Added $manage_packages to control if module install packages

hickey opened the pull request:
This allows one to use the puppet module but not have it install any of the puppet modules. I am championing this as we like the module, but can not handle having packages installed from an OS source as our puppet components are newer versions installed from gems. 

The default behavior out of the box is still to allow the module to manage package installation. This way existing functionality is not broken and still handles according to any existing documentation.

ekohl commented on manifests/init.pp:
```diff
@@ -537,11 +543,15 @@
   validate_re($server_implementation, '^(master|puppetserver)$')
   validate_re($server_parser, '^(current|future)$')
-
+  
```
Nitpick: trailing whitespace

ekohl commented:
I think it's a good change. Please fix the whitespace nitpick, fix the tests and squash the commits.

hickey commented:
OK. Fixed the problem I had. Actually was not the tests, but at least the tests exposed the problem so it could be corrected before it got released. Yea!!! 

Also fixed the extra whitespace in init.pp and squashed the commits into a single commit. 

Thanks.

ekohl commented on manifests/init.pp:
```diff
@@ -537,11 +543,15 @@
+  if $manage_packages != true and $manage_packages != false {
+    validate_re($manage_packages, '^(server|agent)$')
+  }
+  
```
Nitpick: trailing whitespace.

ekohl commented:
:+1: the test failure is unrelated (stdlib master appears to have broken 2.7 compatibility). It'd be great if you could remove the two instances of trailing whitespace but I can also cherry-pick the commit and fix it there.

mmoll commented:
thanks @hickey!
